src/V3Ast.h

```diff
@@ -1719,7 +1719,7 @@ class AstNode VL_NOT_FINAL {
     // Data type locators
     AstNodeDType* findBitDType() { return findBasicDType(VBasicDTypeKwd::LOGIC); }
     AstNodeDType* findDoubleDType() { return findBasicDType(VBasicDTypeKwd::DOUBLE); }
-    AstNodeDType* findStringDType() { return findBasicDType(VBasicDTypeKwd::STRING); }
+    AstNodeDType* findStringDType() const { return findBasicDType(VBasicDTypeKwd::STRING); }
```

Comment:
Do we need this change in this PR? If not, let's just leave it non-const. This fix should be a separate PR.

Reply:
This is necessary, because it is used in `const` context 5 times (every time we call `findStringDType` here).  In hindsight, it might also be necessary to apply the "must be integral" check every time we call `findStringDType`, because it is related to typing the array index.